lib/imports/api/collab-editor/client/collab-editor.js

```diff
@@ -0,0 +1,138 @@
+import { Meteor } from 'meteor/meteor';
+import { ProseMirror } from './../../../prosemirror/dist/edit'
+import './../../../prosemirror/dist/inputrules/autoinput'
+import './../../../prosemirror/dist/menu/tooltipmenu'
+import './../../../prosemirror/dist/menu/menubar'
+import './../../../prosemirror/dist/collab'
+import { defaultSchema } from './../../../prosemirror/dist/model/defaultschema';
+import { Step } from './../../../prosemirror/dist/transform';
+
+import { getLatestDocSnapshot } from './../../documents/both/methods';
+
+// share a global streamerClient across all CollabEditors
+const streamerClient = new Meteor.Streamer('prosemirror-steps');
+
+export default class CollabEditor {
+  constructor({ docId, proseMirrorOptions }) {
+    this.docId = docId;
+    this.proseMirrorOptions = proseMirrorOptions;
+    this.streamerClient = streamerClient;
+    this.setupEditor();
+  }
+
+  /**
+  * Gets the latest doc state and creates an editor with it.
+  */
+  setupEditor() {
+    let self = this;
+
+    // tell the AuthorityManager to open the doc
```

Comment:
when a new ProseMeteorEditor is created, the client calls `ProseMeteor.openDoc`. this will make the AuthorityManager create a new Authority if there isn't already one. when the callback is complete, we know the server has set up the Authority, so we can request the latest doc state from it and create the editor with that doc state